TheMET/TheMET/ViewModels/CultureArtsViewModel.swift

```diff
@@ -0,0 +1,319 @@
+//
+//  CultureArtsViewModel.swift
+//  TheMET
+//
+//  Created by Анна Ситникова on 01/03/2024.
+//
+
+import Foundation
+import UIKit
+import Combine
+import MetAPI
+
+enum CultureArtState {
+    
+    struct LoadedArtState {
+        let objectID: Int
+        let artistsName: String?
+        let titleText: String?
+        let dateText: String?
+        let mediumText: String?
+        var isLiked: Bool
+        fileprivate(set) var imageStatus: LoadingStatus<UIImage>
+    }
+    
+    case placeholder(artID: ArtID)
+    case loaded(artState: LoadedArtState)
+}
+
+class CultureArtsViewModel {
+    
+    @Published private(set) var cultureArtStatesList: LoadingStatus<[CultureArtState]>
+    @Published private(set) var searchText: String
+    private let presentingControllerProvider: () -> UIViewController?
+    
+    private let metAPI = MetAPI()
+    
+    private var imageLoader: ImageLoader = ImageLoader()
+    
+    private let favoriteService = FavoritesService.standart
+    
+    private var favoriteServiseSubscriber: AnyCancellable?
+    
+    private let culture: String?
+    
+    private var loadingArtIds: [ArtID] = []
+    
+    private var cultureArtsList: [CultureArtState] = []
+    
+    private var isSearchFailed: Bool = false
+    
+    private let departmentIDs: [Int] = [11, 15, 21]
+    
+    private var artsWithImageLoading:[Int] = []
+    
+    private var loadedArts:[Art] = []
+    
+    init(culture: String?, presentingControllerProvider: @escaping () -> UIViewController?) {
+        self.cultureArtStatesList = .loading
+        self.presentingControllerProvider = presentingControllerProvider
+        self.searchText = ""
```

Comment:
The `searchText` is set to empty string at this line and is not ever changed after